### DISCUSSION TITLE
ID Does Not Exist

### DISCUSSION THREAD

**Discussion by ridgedale** [rustdesk/rustdesk#1459] [Q&A]
Just installed rustdesk on another PC today. An ID was automatically generated as usual, but when I try to connect to it via the public servers the following error is returned:

ID does not exist

Other existing installations have been accessible today without any issue, btw.

Do the public servers need to be rebooted?

**Comment by deve0123** [comment]
Same problem in all the machines. All showing the same message Connection error "ID does not exist".

**Comment by porteusconf** [comment]
Not so sure it is a server issue... maybe a client issue as well or instead? Reason I say this is I think I have seen "ID not found" and/or other issues like client hanging at "generating ID" if more than one copy of rustdesk client is running and/or hanging. I think upon launch, rustdesk should check to see if there are already existing rustdesk process(es), and just try to open the existing process. If it sees more than one processes already running, or there is a timeout trying to open existing process, it should suggest a reboot, and/or to try these commands in the Terminal (in MacOS or Linux):

``` shell
  pkill  rustdesk
  sleep 5
  pkill  -9  rustdesk
  sleep 5
echo "    Now relaunch rustdesk  ... Or on linux, just type:   rustdesk       "
echo "    On macos:    open   /Applications/RustDesk.app  "
echo "    If it fails to launch on Mac, then try this command in Terminal "
echo "        /Applications/RustDesk.app/Contents/MacOS/rustdesk "
   /Applications/RustDesk.app/Contents/MacOS/rustdesk
```
If launched from Terminal on macos, then you have to give the Terminal same permissions as Rustdesk for viewing screen, etc (in SysPrefs->Security->Privacy).

**Comment by imonlygitting** [comment]
having same issue here - both win11 boxes:
![Picture1](https://github.com/rustdesk/rustdesk/assets/57381651/2dc617e9-3260-4bd8-ae9c-a31ea896c525)

but it seems the error is one direction only! not using self hosted server... any idea? tried uninstall but seems the settings are persistent as new install sees the same saved targets

**Reply by imonlygitting** [reply]
above was problem with rustdesk from pc B to pc A - any devs with ideas how to fix this? 

i can rustdesk pc A from pc C , but not from pc B to pc A. Tested going from C to B also, and have reinstalled app with same error as above... 

another way to fix might be completely remove the rustdesk data (but how) as it seems reinstalling brings back old settings despite having first uninstalled on pc B - any thoughts on this might also help fix the problem...

thank you in advance

**Comment by Joghurt** [comment]
Same problem

**Reply by Naylatirado94** [reply]
Lo solucionaste? Como!!!

**Comment by MikeFalcor** [comment]
OK - So I ran into this issue when I replaced my server. Not guaranteed to work in your case, but I found that restarting the RustDesk service on the problem PC (the one you can't connect to) does the trick.

My guess is that IDs are only registered when the service starts. I will hope that they change that so that every time the config is updated (settings change), it will re-register with the server. This would prevent a lot of headache and force reboots of servers...

**Comment by bigretromike** [comment]
Sometimes if there is "ID does not exist" its because You or the other Side is on a different Server.

**Comment by cchai-newtrek** [comment]
I have the same issue connect from pc A -> pc B, so in pc B, after I click "Start" in the General Settings -> Service, then I can connect 
![image](https://github.com/rustdesk/rustdesk/assets/164280909/de904a32-38e6-4bee-ad4b-de8a104044ca)

**Comment by Maicky02** [comment]
Hola, la solución a este problema es ir a la sección ajustes del rustdesk, luego ir a la sección red y eliminar lo que sale escrito en el servidor y en la key luego dar en aplicar y listo , esta solución me funciono a mi, espero sea de su ayuda....

**Reply by ranita2005** [reply]
Hice esto y me funciono, solo borre todo lo que etsaba escrito y listo. Gracias

**Comment by maw-it** [comment]
Cutt-off the old signal-server and all clients register to the new one!

That was the trick in my situation. I changed the DNS IP (A) record for the signal-server and wait... but no client ID exists on the new server until a rustdesk-service restart on the client. The real problem: the old signal-server was still online (IP and Ports) and that's why no client reconnect/register to the new server.

**Comment by trbtm** [comment]
I have the same issue. It seems to be a client issue since self hosted and public servers both gave the same result.

Client A (macOS 14.5 aarch64) can not connect to client B (macOS 12.7.5, x86/64) but client B can connect to client A.

**Comment by bhotting** [comment]
I had the same issue, what worked for me is the following.
I’m using version 1.2.3-2
Self hosted.
And you are right its a Client issue, so all below is on the client.

1  I removed the ID server (ip address) and key, so removed my self hosted server info on the client.
2  Stopped the (client) server via general tab
3  Closed the rustdesk client software
4  I renamed the  Rustdesk.toml file ( C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config) So software couldn’t find it anymore (not sure if this step is needed)
5  Started the client software and restarted the server.  
    So it had to connect to the public server
    And the ID changed.
6  Then put back my own server info and it worked.

I use the portable client, if you have installed it, you may have to stop the rustdesk service(?)

Hope this helps you.

**Comment by waivivian** [comment]
I also face this issue when I first downloaded this software. I solved it by the two steps below

<h1> 1.  Change the Network Setting</h1>
I think in the "Network" under the Settings in **both** client and server computers, you should clear all these entries (ID server, Key) (if they are filled originally) and click apply. 

![image](https://github.com/user-attachments/assets/36b92793-b01d-427a-a915-8e5a1a808023)

<h1> 2.  Start the service</h1>
Also, you may notice at the bottom of the homepage (for **both** client and server computers) if there is the  sentence "Service is not running". 

![image](https://github.com/user-attachments/assets/a9337b8b-8d20-4170-a23d-16947f5973ce)

You can start the service by either clicking "Start service" or go to "General" under Settings to click Start as below
![image](https://github.com/user-attachments/assets/9580b20c-1b43-45ce-a411-b03b88c27b2c)

<h2> Note: How to go to Setting page</h2>
Note that Settings page can be open by clicking the vertical ellipsis button located on the right side of your ID in the Home page, as shown below

![image](https://github.com/user-attachments/assets/36f66a63-385c-4541-822f-8defa35c477e)

I'm not sure which of these two steps did the trick, but anyway, it finally works for me! :)

**Reply by rustdesk** [reply]
Good job

**Reply by ep0chzero** [reply]
So friend of mine installed RustDesk from the unofficial website rustdesk(dot)io. We encountered an issue where the "ID not found" error appeared. Upon investigation, we found that the unofficial version had configured a custom ID/Relay server pointing to api.rustdesk.io with an API key.

Even after uninstalling the unofficial version and reinstalling RustDesk from the official source, the custom relay server settings persisted. This caused ongoing connection issues, even though the app was now from the official source.

It would be great if RustDesk could mitigate this issue perhaps by clearing or verifying relay server configurations during installation. This would be a valuable security improvement to prevent such persistence from unofficial sources. So theres a way to mitigate this issue and prevent it from happening again.

**Comment by blackw4ll** [comment]
I've been having trouble recently with this as well but have fixed mine with just the simplest way possible. I think one of the issues is that the IDs have a different format when typing it in.

Like PC 1's ID is like 12 345 678 900 (11 digits), and PC 2 is 123 456 789 (9 digits). When I type in PC 2's ID it follows the pattern/format of PC 1's ID like it would be 12 345 678 9 and it shows it does not exist.

So I copied the ID from PC 2 so it should say 123 456 789 and it connected. Hope this makes sense and helps anybody.

**Comment by naddel91** [comment]
Same problem here, I even copied the ID to .txt and pasted it on another machine. "ID does not exist" which is clearly not true.
Why is that?

**Comment by qingqizeng** [comment]
same as mine, the problem happend on a 8 digs ID, i can not figure out how this ID be generted, or how to generate another ID to solve the problem
version 1.3.0 newly installed.. not fixed
self-hosted server: not works
public server: not works
tried 5 different client terminals, 4 out of 5 are ok, only one shows ID not existed
problem client setup on PVE8.0.2'/Ubuntu 22.4 VM
shows connected but other VM or Terminals or Mac can not rust desk to it .... ID not existed

update:
reinstalled then fixed with a new ID (still 8 digs)

**Comment by ep0chzero** [comment]
So friend of mine installed RustDesk from the unofficial website rustdesk(dot)io. We encountered an issue where the "ID not found" error appeared. Upon investigation, we found that the unofficial version had configured a custom ID/Relay server pointing to api.rustdesk.io with an API key.

Even after uninstalling the unofficial version and reinstalling RustDesk from the official source, the custom relay server settings persisted. This caused ongoing connection issues, even though the app was now from the official source.

It would be great if RustDesk could mitigate this issue perhaps by clearing or verifying relay server configurations during installation. This would be a valuable security improvement to prevent such persistence from unofficial sources. So theres a way to mitigate this issue and prevent it from happening again.

**Comment by meilimaoxyn** [comment]
I think I've also encountered this problem. My Windows client is fine. I'm not quite sure if my situation is the same as yours?
